### PR TITLE
Integrate rush git hooks support

### PR DESCRIPTION
**Pull request by natalieethell**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ npm run change`

#### Description of changes

Removes the dependency on husky and instead utilizes the git hooks support added in Rush version 5.5.1. 

"rush install" will copy all of the scripts in `common/git-hooks` over to the `.git/hooks` folder of your local working folder.

More documentation here: https://rushjs.io/pages/maintainer/git_hooks/.

#### Focus areas to test

Pull these changes in locally, make a change to a file, and verify that the pre-commit hook runs.

### CODE REVIEW

File: common/git-hooks/pre-commit [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
```
Comment (ecraig12345) [inline common/git-hooks/pre-commit]: Does this work on Windows even if you don't have sh on your path (from git or some other source)?

Comment (natalieethell) [reply common/git-hooks/pre-commit]: Just spoke with Pete offline and he says yes. He believes that git should sort that out before it invokes the hook

**Comment by natalieethell** [comment]
Just one note - I'll wait until the next version of rush is published to merge this PR, which will include [this fix ](https://github.com/Microsoft/web-build-tools/pull/980) to assign read permissions to the hooks

File: common/git-hooks/pre-commit (hunk 1 above)
Comment (kenotron) [reply common/git-hooks/pre-commit]: This blog seems like it wouldn't work: https://www.tygertec.com/git-hooks-practical-uses-windows/ - but that's back in 2016. Maybe they fixed this?

Comment (natalieethell) [reply common/git-hooks/pre-commit]: What do you think @pgonzal?

Comment (pgonzal) [reply common/git-hooks/pre-commit]: As far as I know, Git Hooks work fine on Windows.

**Comment by pgonzal** [comment]
Awesome!

@johnguy0 @iclanton @lijunle We should consider setting up Prettier in the sp-client repo.

**Pull request merged**